### PR TITLE
Forward projections templates

### LINKED ISSUE #31: Add forward projections import template
Add import template which can be used for forward projections (for 3 years ahead), for each fiscal year.

### PR DESCRIPTION
**Pull request by markbrough**
This PR includes the following:

- export MTEF projections templates, including pre-populated with summed data for each FY, and validation
- import MTEF projections, processing MTEF projections, activity status, start and end dates
- handle also activity status, start and end dates for disbursements templates
- fix a couple of places where years were hardcoded into the system (there is still some of this, however)
- fixes to a couple of small bugs

Fixes #31 and most of #30

### CODE REVIEW

File: maediprojects/query/generate_xlsx.py [hunk 1]
```diff
@@ -78,9 +79,9 @@ def __init__(self, headers):
 def tidy_amount(amount_value):
     amount_value = amount_value.strip()
     amount_value = re.sub(",", "", amount_value)
-    if re.match("^\d*$", amount_value): # 2000
+    if re.match("^-?(\d*)$", amount_value): # 2000 or -2000
         return (float(amount_value), u"USD")
-    elif re.match('(\d*\.\d*)', amount_value):
+    elif re.match('-?(\d*\.\d*)', amount_value):
```
Comment (andylolz) [inline maediprojects/query/generate_xlsx.py]: I think you can do:
```regex
-?(\d*\.?\d*)
```
…and then you can merge the two regexes together.

File: maediprojects/query/generate_xlsx.py [hunk 2]
```diff
@@ -276,20 +373,46 @@ def generate_xlsx_export_template(data):
...
-        for rownum in range(1+1, len(activities)+2):
-            writer.ws.cell(row=rownum,column=4).fill = myFill
-            writer.ws.cell(row=rownum,column=4).number_format = u'"USD "#,##0.00'
-            #writer.ws.cell(row=rownum,column=4).protection = Protection(locked=False)
-        v_id.add('A2:A{}'.format(len(activities)+2))
-        v_number.add('D2:D{}'.format(len(activities)+2))
-        v_status.add('E2:E{}'.format(len(activities)+2))
-        v_date.add('F2:G{}'.format(len(activities)+2))
+            existing_activity = activity_to_json(activity, cl_lookups)
+            for mtef_year in mtef_cols:
```
Comment (andylolz) [inline maediprojects/query/generate_xlsx.py]: Seems like there’s a problem here. `mtef_cols` is only set if `mtef` is True, which is only the case for the MTEF projections reporting. If I try and download a template for the quarterly disbursement reporting, I get a “`mtef_cols` referenced before assignment” error.

Comment (andylolz) [inline maediprojects/query/generate_xlsx.py]: Seems like there’s a problem here. `mtef_cols` is only set if `mtef` is True, which is only the case for the MTEF projections reporting. If I try and download a template for the quarterly disbursement reporting, I get a “`mtef_cols` referenced before assignment” error.

File: maediprojects/query/generate_xlsx.py [hunk 3]
```diff
@@ -234,8 +323,16 @@ def generate_xlsx_filtered(arguments):
...
-    _headers = [u"ID", u"Project code", u"Activity Title", util.previous_fy_fq(),
+def generate_xlsx_export_template(data, mtef=False):
+    if mtef:
+        current_year = datetime.datetime.utcnow().date().year
+        mtef_cols = [u"FY{}/{} (MTEF)".format(str(year)[2:4], str(year+1)[2:4]) for year in range(current_year, current_year+3)]
+        _headers = [u"ID", u"Project code", u"Activity Title"]
+        _headers += mtef_cols
+        _headers += [u'Activity Status', u'Activity Dates (Start Date)', u'Activity Dates (End Date)',
+    u"County"]
+    else:
```
Comment (andylolz) [inline maediprojects/query/generate_xlsx.py]: maybe just here there should be a:

```python
    else:
        mtef_cols = []
```
…sort of thing.

File: maediprojects/query/generate_xlsx.py (hunk 2 above)
Comment (andylolz) [reply maediprojects/query/generate_xlsx.py]: I guess https://github.com/bsi-liberia/liberia-projects/pull/39/files#r266686772 would fix this.

Comment (markbrough) [reply maediprojects/query/generate_xlsx.py]: Thank you! This actually made me realise there was some remaining stuff to be done in the disbursement import, to also process the new columns in the disbursement export template (e.g. activity status, etc.). Also, I guess this function could generally be much tidier...

File: maediprojects/query/generate_xlsx.py (hunk 3 above)
Comment (markbrough) [reply maediprojects/query/generate_xlsx.py]: Should be fixed in c5ee807

**Comment by andylolz** [comment]
Oops – good catch in d09d5a6. Sorry about that!

**Comment by markbrough** [comment]
Hurray!

**Pull request merged**